openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py

```diff
@@ -0,0 +1,301 @@
+    # pylint: disable=redefined-outer-name
+    """
+    Test to check that a user with two active enrollments in a total of two prerequisites and one enrollment in a
+    regular course is counted and a user with no enrollments is not counted while calculating minimal users
+    """
+    prerequisites, user = enroll_users
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(prerequisites)
+
+    assert minimal_users.count() == 1
+    assert minimal_users.first().username == user.username
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('create_users', [2], indirect=True)
+@pytest.mark.parametrize('create_courses', [[True, False, True]], indirect=True)
+@pytest.mark.parametrize('enroll_users', [[True, True]], indirect=True)
+def test_get_minimal_users_to_be_checked_for_update_with_1_active_enrollment_in_prereq_and_1_in_course(enroll_users):
+    # pylint: disable=redefined-outer-name
+    """
+    Test to check that a user with one active enrollments in a total of two prerequisite and one enrollment in a
+    regular course and a user with no enrollments are not counted while calculating minimal users
+    """
+    prerequisites, _ = enroll_users
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(prerequisites)
+
+    assert minimal_users.count() == 0
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('create_users', [2], indirect=True)
+@pytest.mark.parametrize('create_courses', [[True, True, True]], indirect=True)
+@pytest.mark.parametrize('enroll_users', [[True, True]], indirect=True)
+def test_get_minimal_users_to_be_checked_for_update_with_2_active_enrollments_out_of_3_prereqs(enroll_users):
+    # pylint: disable=redefined-outer-name
+    """
+    Test to check that a user with two active enrollments in a total of three prerequisite and a user with no
+    enrollments are not counted while calculating minimal users
+    """
+    prerequisites, _ = enroll_users
+
+    command = command_module.Command()
+    minimal_users = command.get_minimal_users_to_be_checked_for_update(prerequisites)
+
+    assert minimal_users.count() == 0
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('create_users', [2], indirect=True)
+@pytest.mark.parametrize('create_courses', [[True, True, True]], indirect=True)
+@pytest.mark.parametrize('enroll_users', [[True, True]], indirect=True)
+def test_get_minimal_users_to_be_checked_for_update_case_with_prerequisites_already_passed(enroll_users):
+    # pylint: disable=redefined-outer-name
+    """
+    Test to check that a user already passed in prerequisites and a user with no enrollments are not counted while
+    calculating minimal users
+    """
+    prerequisites, user = enroll_users
+
+    user_application_hub, _ = ApplicationHub.objects.get_or_create(user=user)
```

Comment:
Lets create a factory for application hub using factory boy and use [get_or_create](https://factoryboy.readthedocs.io/en/2.12.0/orms.html#factory.django.DjangoOptions.django_get_or_create)